lambda/tests/test_new_account_trust_policy.py

```diff
@@ -0,0 +1,370 @@
+    with mock_sts():
+        yield boto3.client("sts", region_name=AWS_REGION)
+
+
+@pytest.fixture(scope="function")
+def org_client(aws_credentials):
+    """Yield a mock organization that will not affect a real AWS account."""
+    with mock_organizations():
+        yield boto3.client("organizations", region_name=AWS_REGION)
+
+
+@pytest.fixture(scope="function")
+def mock_event(org_client):
+    """Create an event used as an argument to the Lambda handler."""
+    org_client.create_organization(FeatureSet="ALL")
+    account_id = org_client.create_account(
+        AccountName=MOCK_ORG_NAME, Email=MOCK_ORG_EMAIL
+    )["CreateAccountStatus"]["Id"]
+    return {
+        "version": "0",
+        "id": str(uuid.uuid4()),
+        "detail-type": "AWS API Call via CloudTrail",
+        "source": "aws.organizations",
+        "account": "222222222222",
+        "time": datetime.now().isoformat(),
+        "region": AWS_REGION,
+        "resources": [],
+        "detail": {
+            "eventName": "CreateAccount",
+            "eventSource": "organizations.amazonaws.com",
+            "responseElements": {
+                "createAccountStatus": {
+                    "id": account_id,
+                }
+            },
+        },
+    }
+
+
+@pytest.fixture(scope="session")
+def initial_trust_policy():
+    """Return AssumeRolePolicyDocument used when creating a role."""
+    initial_json = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Action": "sts:AssumeRole",
+                "Principal": {"AWS": f"arn:aws:iam::{ACCOUNT_ID}:root"},
+                "Effect": "Allow",
+            }
+        ],
+    }
+    return jsonpickle.encode(initial_json)
+
+
+@pytest.fixture(scope="session")
+def replacement_trust_policy():
+    """Return JSON policy used for updating the AssumeRolePolicyDocument."""
+    arn = f"arn:aws:iam::{ACCOUNT_ID}:saml-provider/saml-provider"
+    valid_json = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Action": "sts:AssumeRole",
+                "Principal": {"AWS": f"arn:aws:iam::{ACCOUNT_ID}:root"},
+                "Effect": "Allow",
+            },
+            {
+                "Action": "sts:AssumeRoleWithSAML",
+                "Principal": {
+                    "Federated": arn,
+                },
+                "Effect": "Allow",
+            },
+        ],
+    }
+    return json.dumps(valid_json)
+
+
+def reset_roles(
+    iam_client, trust_policy, role_name_list
+):  # pylint: disable=redefined-outer-name
+    """Create role(s) with the same initial AssumeRolePolicyDocument.
+
+    If the roles already exist, they will be deleted.
+    """
+    role_names = [role["RoleName"] for role in iam_client.list_roles()["Roles"]]
+
+    for role_name in set(role_name_list):
+        if role_name in role_names:
+            iam_client.delete_role(RoleName=role_name)
```

Comment:
This is setting off my spidey sense as a possible anti-pattern or sign there is some other issue with the test setup... Isn't the moto state cleared between tests? I thought the decorators and context managers were _supposed_ to clear the state...

Reply:
Ok

Reply:
It was really a question, not a directive. If there's a reason for deleting the role, then sure. It just feels off to do it in this particular manner. Perhaps the iam role should be created and deleted with a fixture?

Reply:
The role names are unique per test.